### PR TITLE
[TF-14334] Make Service Worker ready

### PR DESCRIPTION
**Pull request by PabloDotcom**

### CODE REVIEW

File: dist/timesync.js [hunk 1]
```diff
@@ -862,7 +862,11 @@ Promise.disableSynchronous = function() {
 },{"./core.js":4}],11:[function(require,module,exports){
 'use strict';
 
-module.exports = typeof window === 'undefined' || typeof window.Promise === 'undefined' ? require('promise') : window.Promise;
+var selfPromise = typeof self !== 'undefined' && typeof self.Promise !== 'undefined' && self.Promise;
```
Comment (reisandbeans) [inline dist/timesync.js:865]: quick question: why do we need this? All major browsers have promises, and node has it well. couldn't we simply use the global Promise (available both in the window and the node global object)?

File: dist/timesync.js [hunk 2]
```diff
@@ -920,75 +924,36 @@ function emitter(obj) {
...
-        return reject(err);
-      }
-
-      resolve([res, status]);
-    };
-
-    fetch('POST', url, body, null, callback, timeout);
-  });
+    return new Promise(function (resolve, reject) {
+        return Promise.race([fetch(url, {
```
Comment (reisandbeans) [inline dist/timesync.js]: if you're already returnin a Promise.race, why do you need to wrap it with a new promise? Also, can we use arrow functions?

File: lib/Promise.js [hunk 3]
```diff
@@ -1,3 +1,7 @@
 'use strict';
 
-module.exports = typeof window === 'undefined' || typeof window.Promise === 'undefined' ? require('promise') : window.Promise;
\ No newline at end of file
+var selfPromise = typeof self !== 'undefined' && typeof self.Promise !== 'undefined' && self.Promise;
```
Comment (reisandbeans) [inline lib/Promise.js:3]: Same question as above

File: lib/request/request.browser.js [hunk 4]
```diff
@@ -1,64 +1,28 @@
...
+            redirect: 'follow',
+            body: JSON.stringify(body)
+        }).then(function (response) {
+            return response.json().then(function (respJson) {
+                return resolve([respJson, response.status]);
+            });
+        }).catch(function (err) {
+            return reject(err);
+        }), new Promise(function (_, reject) {
+            return setTimeout(function () {
```
Comment (reisandbeans) [inline lib/request/request.browser.js]: btw, you can use AbortControllers to simulate a timeout: https://dmitripavlutin.com/timeout-fetch-request/

**Review by reisandbeans — COMMENTED** [review]
since we're changing this lib, I wonder if it doesn't make more sense to create two exports from this package instead of the multiple `if browser/node, use this, else use that` and have the browser require one, and node require the other. See https://stackoverflow.com/a/63058817

**Review by rhaylander — COMMENTED** [review]
Please, see comments form @Tinadim

File: dist/timesync.js (hunk 1 above)
Comment (PabloDotcom) [reply dist/timesync.js:865]: no idea, wanted to keep backward compatibility.

File: dist/timesync.js (hunk 2 above)
Comment (PabloDotcom) [reply dist/timesync.js]: to have access to resolve and reject?

File: lib/Promise.js (hunk 3 above)
Comment (PabloDotcom) [reply lib/Promise.js:3]: This and the above is dist of src/Promise.js.

**Review by PabloDotcom — COMMENTED** [review]
It's not my intention to change entire lib @Tinadim @rhaylander. I just want it service worker ready.

File: dist/timesync.js (hunk 2 above)
Comment (PabloDotcom) [reply dist/timesync.js]: There is no arrow functions in the rest of this project.

File: lib/request/request.browser.js (hunk 4 above)
Comment (PabloDotcom) [reply lib/request/request.browser.js]: nice one! thanks, changed!

File: dist/timesync.js (hunk 2 above)
Comment (reisandbeans) [reply dist/timesync.js]: The rest of tenfold code base uses arrow functions. Plus every major browser and node version support it now. We should follow our own standards

File: dist/timesync.js (hunk 1 above)
Comment (reisandbeans) [reply dist/timesync.js:865]: I don't think that should be a concern. Like I said, all major browsers support promises now.

**Comment by reisandbeans** [comment]
@PabloDotcom understood, but if you're touching the code and changing lines, you should make sure that the lines you changed follow our best practices. If you're running out of time and need to deliver, then we can create a ticket to fix that in a next sprint. Otherwise we need to make sure that at least the lines you've changed are updated to our most recent standards.

**Review by louishawkins — APPROVED** [review]

**Review by danego — APPROVED** [review]

**Comment by louishawkins** [comment]
Created a task to address comments https://growtenfold.atlassian.net/browse/TF-15966

**Pull request merged**